### PR TITLE
Feat/4044 delete topic

### LINKED ISSUE #4018: Ability to remove Topics
### Feature description

Currently, we are only able to create topics but not remove them. This is an essential feature.

image

### Requirements:

- Allow the user to delete a Topic
- if the topic is not connected to any (receiver/sender) entity it can just be deleted
- If the topic being deleted is currently connected to any service (receiver/sender) show the user a warning before actually deleting the topic
    - "Are you sure?" notification design will be provided
    - if the user cancels the request he is returned to the page without deleting the topic
    - if the user approves the request the topic is deleted

Notice - any existing services connected to the topic will lose the connection configuration once the topic is deleted.

### Design
https://www.figma.com/file/ZYExoG89CFPibfoQHq4gpB/%234018---Topic---Are-you-sure%3F?node-id=1136%3A6535&t=Wd47IWspGq2ZJVl6-1

### How to Demo:
**Scenario 1 - Without dependancy**
1. Create a new topic
2. Delete the topic
3. The topic is deleted

**Scenario 2 - With dependancy**
1. Create a new topic
2. Connect the topic to several services
3. Try to delete the topic
4. Get an "Are you sure ?" notification with info on connected services
5. cancel the request
6. Verify the topic wasnt deleted
7. Try to delete the topic
4. Get an "Are you sure ?" notification with info on connected services
5. approve the request
6. The topic is deleted
7. All connections configured for the topic are deleted

### PR DESCRIPTION
**Pull request by morhag90**
Close #4018 

## PR Details

Add delete topic in message broker resource 

## PR Checklist
- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

**Comment by arielweinberger** [comment]
Let's wait with merging this until we test the new Nx React setup, just to minimize chances errors introduced.

File: src/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,1609 @@
+# ------------------------------------------------------
+# THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
+# ------------------------------------------------------
+
+type Auth {
+  """
+  JWT Bearer token
+  """
+  token: String!
+}
```
Comment (yuval-hazaz) [inline src/schema.graphql:10]: remove this file

File: packages/amplication-server/src/core/topic/topic.service.ts [hunk 2]
```diff
@@ -14,11 +16,15 @@ export class TopicService extends BlockTypeService<
...
-  UpdateTopicArgs
+  UpdateTopicArgs,
+  DeleteTopicArgs
 > {
   blockType = EnumBlockType.Topic;
 
-  constructor(protected readonly blockService: BlockService) {
+  constructor(
+    protected readonly blockService: BlockService,
+    protected readonly serviceTopics: ServiceTopicsService
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/topic/topic.service.ts:26]:
```suggestion
    protected readonly serviceTopicsService: ServiceTopicsService
```

File: packages/amplication-server/src/core/topic/topic.module.ts [hunk 3]
```diff
@@ -3,10 +3,18 @@ import { TopicService } from "./topic.service";
...
-  providers: [TopicService, TopicResolver],
-  exports: [TopicService, TopicResolver],
+  imports: [
+    BlockModule,
+    PermissionsModule,
+    ServiceTopicsModule,
+    ResourceModule,
+  ],
+  providers: [TopicService, TopicResolver, ServiceTopicsService],
+  exports: [TopicService, TopicResolver, ServiceTopicsService],
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/topic/topic.module.ts:18]:
```suggestion
  exports: [TopicService, TopicResolver],
```

File: packages/amplication-server/src/core/topic/topic.module.ts [hunk 4]
```diff
@@ -3,10 +3,18 @@ import { TopicService } from "./topic.service";
...
-  imports: [BlockModule, PermissionsModule],
-  providers: [TopicService, TopicResolver],
-  exports: [TopicService, TopicResolver],
+  imports: [
+    BlockModule,
+    PermissionsModule,
+    ServiceTopicsModule,
+    ResourceModule,
+  ],
+  providers: [TopicService, TopicResolver, ServiceTopicsService],
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/topic/topic.module.ts:17]:
```suggestion
  providers: [TopicService, TopicResolver],
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
in addition to the below - we need to throw exceptions in the settings blocks so they will not be deletable

File: packages/amplication-server/src/core/topic/topic.module.ts (hunk 3 above)
Comment (morhag90) [reply packages/amplication-server/src/core/topic/topic.module.ts]: removed

File: packages/amplication-client/src/Topics/Topic.tsx [hunk 5]
```diff
@@ -1,25 +1,30 @@
...
+const CLASS_NAME = "topic";
 
 const Topic = () => {
   const match = useRouteMatch<{
     resource: string;
     topicId: string;
   }>("/:workspace/:project/:resource/topics/:topicId");
 
-  const { topicId } = match?.params ?? {};
+  const { topicId, resource } = match?.params ?? {};
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/Topics/Topic.tsx:25]: history

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**